scripts/test/support.py

```diff
@@ -145,7 +145,7 @@ def to_end(j):
             ret += [(chunk, [])]
         elif chunk.startswith('(assert_invalid'):
             continue
-        elif chunk.startswith(('(assert', '(invoke')):
+        elif chunk.startswith(('(assert', '(invoke', '(register')):
```

Comment:
This seems to assume that the format of spec tests is "a module, then some more registered modules for that module, plus asserts/invokes". Is that true? It seems like in theory there could be a few modules up front that are registered and then used by multiple modules later for example.

Reply:
The syntax for `register` is simply `(register "a")`, so it always expects at least a module first. And it's often the case that a module is registered first then is used in subsequent modules. So in this line, I simply wanted to include any `(register` commands, so that we can rebuild the spec test with roughly the same structure. It should not assume any specific structure on spec tests more than it was doing before. I may be missing the point, though...

Reply:
Yes, it doesn't assume any more structure, I am just worried it is not enough for future spec tests that need more structure. Specifically, this ends up putting each registered module into one "chunk", but it may be used by multiple ones?

Anyhow, that doesn't seem urgent to figure out in this PR, and maybe my worry is not relevant...

Reply:
I may have misunderstood your comment, but just for the sake of clarification, note that this PR won't run spec tests in chunks anymore. It only run `wasm-opt`, `wasm-as` and `wasm-dis` on chunks, then puts the whole thing together again and run `wasm-shell` once on the resulting `spec.wast`, so that all registered modules are available once again.

Reply:
Thanks! That's what I was missing. I didn't see the higher-up running of all the spec test at once...